File: backend/src/main/java/com/scholarsmanuscript/entity/Article.java
package com.scholarsmanuscript.entity;

import jakarta.persistence.*;
import lombok.*;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;

import java.time.LocalDateTime;
import java.util.HashSet;
import java.util.Set;

@Entity
@Table(name = "articles")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Article {

    public enum Status {
        DRAFT, PUBLISHED
    }

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false, length = 200)
    private String title;

    @Column(columnDefinition = "TEXT")
    private String content;

    @Column(length = 500)
    private String excerpt;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false, length = 20)
    private Status status;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "author_id", nullable = false)
    private User author;

    @OneToMany(cascade = CascadeType.ALL, orphanRemoval = true)
    @JoinColumn(name = "article_id")
    @Builder.Default
    private Set<ArticleCategory> articleCategories = new HashSet<>();

    @CreationTimestamp
    @Column(name = "created_at", updatable = false)
    private LocalDateTime createdAt;

    @UpdateTimestamp
    @Column(name = "updated_at")
    private LocalDateTime updatedAt;

    @Column(name = "published_at")
    private LocalDateTime publishedAt;

    public void addCategory(Category category) {
        ArticleCategory articleCategory = ArticleCategory.builder()
                .articleId(this.id)
                .categoryId(category.getId())
                .build();
        this.articleCategories.add(articleCategory);
    }

    public void removeCategory(Category category) {
        this.articleCategories.removeIf(ac -> ac.getCategoryId().equals(category.getId()));
    }
}

File: frontend/src/types/article.ts
export interface ArticleRequest {
  title: string
  content: string
  excerpt: string
  status: 'DRAFT' | 'PUBLISHED'
  categoryIds?: number[]
}

export interface ArticleResponse {
  id: number
  title: string
  excerpt: string
  status: string
  author: string
  createdAt: string
  publishedAt: string | null
  categories: string[]
  readTime: string
}

export interface ArticleDetailResponse extends ArticleResponse {
  content: string
  authorId: number
  updatedAt: string
}

export interface Post {
  id: number
  title: string
  summary: string
  content: string
  category: string
  author: string
  date: string
  readTime: string
  featured: boolean
}

export interface Category {
  id: string
  name: string
  subItems: string[]
}
